sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala

```diff
@@ -73,15 +94,146 @@ object HiveThriftServer2 extends Logging {
     }
   }
 
+  private[thriftserver] class SessionInfo(
+      val sessionId: String,
+      val startTimestamp: Long,
+      val ip: String,
+      val userName: String) {
+    var finishTimestamp: Long = 0L
+    var totalExecute: Int = 0
+    def totalTime: Long = {
+      if (finishTimestamp == 0L) {
+        System.currentTimeMillis() - startTimestamp
+      } else {
+        finishTimestamp - startTimestamp
+      }
+    }
```

Comment:
According to [this section] [1] of Databricks Scala Guide, I think we prefer `System.nanoTime` to do timing.

[1]: https://github.com/databricks/scala-style-guide#prefer-nanotime-over-currenttimemillis